src/redux/actionTypes.ts

```diff
@@ -1,12 +1,14 @@
+import { InboxItem } from 'models';
+
 export const CHANGE_SECTION = 'change-section';
+export const SELECT_ITEM = 'select-item';
 
 export enum SectionName {
   BugReports = 'Bug Reports',
   CustomerService = 'Customer Service',
   FeatureRequests = 'Feature Requests',
 }
 
-export type AppAction = {
-  type: typeof CHANGE_SECTION;
-  sectionName: SectionName;
-};
+export type AppAction =
+  | { type: typeof CHANGE_SECTION; sectionName: SectionName }
+  | { type: typeof SELECT_ITEM; item: InboxItem | undefined };
```

Comment:
lets change the type to be `item: InboxItem` and only dispatch the action if its not `undefined`

Reply:
if we only dispatch when its not `undefined` then how would the app state's `selectedItem` get get back to `undefined` when the item is unselected?

Reply:
oh right my bad. you're right lets keep it then

Reply:
can i do the question mark here too

Reply:
like `item?: InboxItem` for the parameter

Reply:
good question i am not sure actually. try it and lmk

Reply:
i changed it everywhere i had `Inbox Item | undefined` before and it looks like it works